test/init_fini/src/register_custom_gc.c

```diff
@@ -0,0 +1,6 @@
+extern void register_mygc();
+
+__attribute__((constructor)) static void xxx_ctor()
+{
+    register_mygc();
+}
```

Comment:
Registration still needs a solution on Windows (though something similar is possible). I'd prefer if we would not have to rely on C, though.

Some time ago, I tried implementing `pragma(dataSeg, "segment-name")`, would be pretty helpful for this, too. I think it's badly needed in a system programming language.

Reply:
I was hoping for the _STI/_STD mechanism to work (to be recognized by the linker), but must have misremembered that it works.
There is https://issues.dlang.org/show_bug.cgi?id=16300 and we've recently worked on constructor/destructor functions https://github.com/dlang/dmd/pull/6956 that could be added as pragma(constructor) and pragma(destructor). Not sure what gdc or ldc are doing with regards to that.

Reply:
We have https://github.com/ldc-developers/druntime/blob/ldc/src/ldc/attributes.d#L229 in LDC, with the addition of `pragma(crt_constructor)` and friends, perhaps we should add that?